File: 1534.c
#include <stdio.h>

int main() {
    int n, i, j, aux;

    while(scanf("%d", &n) != EOF) {
        for(i = 0; i < n; i++) {
            for(j = 0; j < n; j++) {
                if (i + j == n - 1)       // 1º: diagonal secundária
                    aux = 2;
                else if (i == j)          // 2º: diagonal principal
                    aux = 1;
                else
                    aux = 3;

            printf("%d", aux);

            }
            printf("\n");
        }
    }

    return 0;
}

File: 1534.py
while True:
    try:
        n = int(input())
        for i in range(n):
            linha = ""
            for j in range(n):
                if i + j == n - 1:      # diagonal secundária
                    aux = 2
                elif i == j:            # diagonal principal
                    aux = 1
                else:                   # resto
                    aux = 3
                linha += str(aux)
            print(linha)
    except EOFError:
        break
